dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNDataInterfaceIIDM.cpp

```diff
@@ -102,21 +113,35 @@ DataInterfaceIIDM::build(std::string iidmFilePath) {
 #endif
     boost::split(paths, extensionsPaths, boost::is_any_of(splitCharacter));
 
-    for (unsigned int i = 0; i < paths.size(); ++i) {
-      std::regex fileRegex(stdcxx::format(".*libiidm-ext-.*\\%1%", boost::dll::shared_library::suffix().string()));
-      powsybl::iidm::ExtensionProviders<powsybl::iidm::converter::xml::ExtensionXmlSerializer>::getInstance().loadExtensions(paths[i], fileRegex);
-    }
+    loadExtensions(paths);
 
-    powsybl::iidm::Network networkIIDM = powsybl::iidm::Network::readXml(boost::filesystem::path(iidmFilePath), options);
+    auto networkIIDM = boost::make_shared<powsybl::iidm::Network>(powsybl::iidm::Network::readXml(boost::filesystem::path(iidmFilePath), options));
 
-    data.reset(new DataInterfaceIIDM(std::move(networkIIDM)));
+    if (nbVariants > 1) {
+      auto& manager = networkIIDM->getVariantManager();
+      manager.allowVariantMultiThreadAccess(true);
+      for (unsigned int i = 0; i < nbVariants; ++i) {
+        const std::string& variantName = std::to_string(i);
+        constexpr bool overwrite = true;
```

Comment:
Please create this one outside of the loop

Reply:
OK